test/distributions/test_constraints.py

```diff
@@ -46,17 +46,24 @@
 ]
 
 
+def is_lapack_available():
```

Comment:
I noticed that we don't run distributions test when pytorch is not compiled with lapack, and I am also fine with us doing the same here for all tests so that newer tests don't accidentally end up calling some function in `torch.linalg` and failing here:

```
if __name__ == '__main__' and torch._C.has_lapack:
    pytest.main([__file__])
```

Reply:
Hey @neerajprad ! Do I understand correctly that the better way to do this would be to exclude distribution tests in `run_test.py` alltogether (like I recently did for another case: #70677) ?

Reply:
Generally, being more specific in skipping is better (and in fact, we can add a `pytest.skip` marker to specific examples like https://github.com/pytorch/pytorch/blob/644835cc381a306df24fdb5ea5e90704eafb4060/test/distributions/test_constraints.py#L104 which is what we also do for CUDA or scipy not being installed).

However, I noticed that we skip all tests in [test_distributions.py](https://github.com/pytorch/pytorch/blob/644835cc381a306df24fdb5ea5e90704eafb4060/test/distributions/test_distributions.py#L5116) if pytorch isn't compiled with lapack. To keep things consistent, I therefore proposed that we do the same here (constraints are a part of distributions module). Otherwise, it is easy to add a test that might call some function in `torch.linalg` (a fair amount of distributions functionality depends on that) and we'll end up breaking this again.

Reply:
@neerajprad I get your idea. Adding the check your suggest as an additional condition to "if __name__.." line will not prevent test runner `run_test.py` to run those tests (I checked). It _will_ prevent however the tests to be run if user by some reason decides to run the module manually. It could be added as an additional prevention measure. But to exclude them from the standard runner we have to work on the level of `run_test.py`'s source code. I refer to one such fix for another case like this in the comment above.

Reply:
@neerajprad The reason for different behaviour in `test_constraints` and `test_distributions` could be (at a first quick look) because the `test_constraints` `[is included](https://github.com/pytorch/pytorch/blob/644835cc381a306df24fdb5ea5e90704eafb4060/test/run_test.py#L168)` in USE_PYTEST_LIST which results in launching it with pytest that imports test and run them from module on its own. `test_distributions` is not (included) in the list and runs probably just as script.

Reply:
Thanks for explaining. 

> As I understand, adding the check your suggest as an additional condition to "if __name__.." line will not prevent test runner run_test.py from running those tests (I checked).

That's interesting. Any ideas why you are only observing these failures in `distributions/test_constraints.py` and not `distributions/test_distributions.py`? My initial thought was that this was because of the additional check after `if __name__` in `test_distributions.py` but your finding seems to suggest that there's some other place where these tests might be getting excluded.

Reply:
I just saw your other comment, that makes sense! In that case, we can go with your solution of removing all distributions tests in `run_test.py` itself.

Reply:
ok, will try to fix it :)